src/Matej.php

```diff
@@ -27,7 +28,7 @@ public function request(): RequestBuilderFactory
     }
 
     /** @return $this */
-    public function setHttpClient(HttpClient $client): self
+    public function setHttpClient(ClientInterface $client): self
```

Comment:
All these new types look like breaking changes. Are you gonna release new major version? If so, there is no minor version with deprecation notices for easier transition?

Reply:
I don't think it would be possible / effective to have this transition path with deprecation notices. IMHO releasing new version with new PHP verison dependencies would be enough...

Reply:
I think, It's not breaking change, because old HttpClient can be still used, becuase it implements this interface - https://github.com/php-http/httplug/blob/2.x/src/HttpClient.php